objects/src/accounts/storage/mod.rs

```diff
@@ -267,20 +276,16 @@ impl AccountStorage {
         // --- update storage maps --------------------------------------------
 
         for &(slot_idx, ref map_delta) in delta.updated_maps.iter() {
-            // layout commitment slot cannot be updated
-            if slot_idx == Self::SLOT_LAYOUT_COMMITMENT_INDEX {
-                return Err(AccountError::StorageSlotIsReserved(slot_idx));
+            for &key in map_delta.cleared_leaves.iter() {
+                self.set_map_item(slot_idx, key, Word::default())?;
             }
 
-            // pick the right storage map and apply delta
-            let storage_map =
-                self.maps.get_mut(&slot_idx).ok_or(AccountError::StorageMapNotFound(slot_idx))?;
-            storage_map.apply_delta(map_delta)?;
+            for &(key, value) in map_delta.updated_leaves.iter() {
+                self.set_map_item(slot_idx, key, value)?;
+            }
```

Comment:
If we bring back `StorageMap::apply_delta()` as mentioned, this could look something like:
```Rust
let new_root = self.maps.get_mut(&index)
    .ok_or(AccountError::StorageMapNotFound(index))?
    .apply_delta(map_delta);
let index = LeafIndex::new(slot_idx.into()).expect("index is u8 - index within range");
self.slots.insert(index, new_root);
```

Reply:
Which means we don't need `AccountStorage::set_map_item` anymore, right?

> Separately, we should also implement AccountStorage::set_map_item() and AccountStorage::get_map_item() methods to mimic the ones in the transaction kernel.

Originally, we wanted to have those methods.